objects/ohitbox/Step_2.gml

```diff
@@ -22,228 +22,400 @@ if (!global.game_paused)
 				
-				// Handle if the opponent is Crouch blocking or not
-				if (collision_list[| i].owner.verticalMoveDir == -1)
-				{
-					collision_list[| i].owner.sprite_index = sRussel_Crouch_Block;
-					collision_list[| i].owner.isCrouchBlocking = true;
-				}
-				else 
+					hasHit = true;
+				} else
+				// Grabbing
+				if (attackProperty.attackType[hitboxID] == eAttackType.GRAB &&
+						collision_list[| i].owner.hitstun < 1 &&
+						collision_list[| i].owner.blockstun < 1 &&
+						collision_list[| i].owner.grounded &&
+						collision_list[| i].owner.state != eState.THROW_TECH &&
+						collision_list[| i].owner.state != eState.JUMPSQUAT &&
+						collision_list[| i].owner.isThrowable)
 				{
-					collision_list[| i].owner.sprite_index = sRussel_Block;
-					collision_list[| i].owner.isCrouchBlocking = false;
-				}
+					// Set the correct states for the attacker and victim
+					owner.state = eState.HOLD;
+					owner.animTimer = 0;
 				
-				// Meter Build - P1 gets 75% meter, P2 gets 50%
-				collision_list[| i].owner.superMeter += floor(attackProperty.meterGain[hitboxID] * 0.5);
-				owner.superMeter += floor(attackProperty.meterGain[hitboxID] * 0.75);
+					collision_list[| i].owner.state = eState.BEING_GRABBED;
+					collision_list[| i].owner.sprite_index = collision_list[| i].owner.CharacterSprites.hurt_Sprite;
+					collision_list[| i].owner.animTimer = 0;
+					collision_list[| i].owner.x = owner.x + (attackProperty.holdXOffset[hitboxID] * owner.image_xscale);
+					collision_list[| i].owner.isShortHopping = false; // Make sure the victim is not using their shorthop fall speed.
+					owner.heldOpponent = collision_list[| i].owner;
 				
-				collision_list[| i].owner.knockbackVel = attackProperty.knockback[hitboxID];
-				owner.pushbackVel = attackProperty.pushback[hitboxID];
-				hasHit = true;
-				collision_list[| i].owner.blockstun = attackProperty.blockstun[hitboxID];
-				collision_list[| i].owner.shuffle = 0;
-				global.hitstop = attackProperty.attackHitstop[hitboxID];
+					// Multiple hitboxes
+					hasHit = true;
+					ds_list_add(collision_list[| i].owner.hitByGroup, attackProperty.group[hitboxID]);
 				
-				// Allow Cancelling
-				owner.cancelable = true;
+					// Depth Sorting
+					owner.depth = -1;
+					collision_list[| i].owner.depth = 0;
 				
-				// Multiple hitboxes
-				ds_list_add(collision_list[| i].owner.hitByGroup, attackProperty.group[hitboxID]);
+					// Reset Frame Advantage Counter
+					oGameManager.frameAdvantage = 0;
 				
-				// Depth Sorting
-				owner.depth = -1;
-				collision_list[| i].owner.depth = 0;
+					var particle = instance_create_layer(collision_list[| i].owner.x, collision_list[| i].owner.y, "Particles", oParticles);
+					with (particle) 
+					{
+						lifetime = 10;
+						sprite_index = sHitEffect;
+						image_xscale = other.owner.image_xscale * -1;
+					}
 				
-				// Reset Frame Advantage Counter
-				oGameManager.frameAdvantage = 0;
+				} else 

```

Comment:
Same thing here. What's with this else? Either remove or move it to be with its if